lib/instance.js

```diff
@@ -754,13 +754,23 @@ module.exports = (function() {
    * @alias updateAttributes
    */
   Instance.prototype.update = function(values, options) {
+    var changedBefore = this.changed() || []
+      , sideEffects
+      , fields;
+
     options = options || {};
     if (Array.isArray(options)) options = {fields: options};
 
     this.set(values, {attributes: options.fields});
 
+    // Now we need to figure out which fields were actually affected by the setter.
+    // We build a somewhat weird argument list array that we can apply to _.without
+    changedBefore.unshift(this.changed());
+    sideEffects = _.without.apply(this, changedBefore);
```

Comment:
`.without.apply(this, [this.changed() || []].concat(changedBefore))` would be great yeah. I think it's a bit more easy to understand but i'm not certain that's true :)